pkg/runner/cluster_k8s.go

```diff
@@ -293,6 +295,131 @@ func (*ClusterK8sRunner) ID() string {
+	})
+	if err != nil {
+		return
+	}
+	if len(res.Items) != 1 {
+		return
+	}
+
+	pod := res.Items[0]
+	if pod.Status.Phase != "Running" {
+		return
+	}
+
+	efsCheck = api.HealthcheckItem{Name: "efs", Status: api.HealthcheckStatusOK, Message: "efs provisioner is running"}
+	return
+}
+
+func (c *ClusterK8sRunner) healthcheckRedis() (redisCheck api.HealthcheckItem) {
+	redisCheck = api.HealthcheckItem{Name: "redis", Status: api.HealthcheckStatusFailed, Message: "redis service is not running"}
+
+	client := c.pool.Acquire()
+	defer c.pool.Release(client)
+
+	res, err := client.CoreV1().Pods(c.config.Namespace).List(metav1.ListOptions{
+		LabelSelector: "app=redis",
+	})
+	if err != nil {
+		return
+	}
+	if len(res.Items) != 1 {
+		return
+	}
+
+	pod := res.Items[0]
+	if pod.Status.Phase != "Running" {
+		return
+	}
+
+	redisCheck = api.HealthcheckItem{Name: "redis", Status: api.HealthcheckStatusOK, Message: "redis service is running"}
+	return
+}
+
+func (c *ClusterK8sRunner) healthcheckSidecar() (sidecarCheck api.HealthcheckItem) {
+	sidecarCheck = api.HealthcheckItem{Name: "sidecar", Status: api.HealthcheckStatusFailed, Message: "sidecar service is not reachable"}
+
+	// get number of worker nodes
+	client := c.pool.Acquire()
+	defer c.pool.Release(client)
+
+	res, err := client.CoreV1().Nodes().List(metav1.ListOptions{
+		LabelSelector: "kubernetes.io/role=node",
+	})
+	if err != nil {
+		return
+	}
+
+	nodes := len(res.Items)
+
+	pods, err := client.CoreV1().Pods(c.config.Namespace).List(metav1.ListOptions{
+		LabelSelector: "name=testground-sidecar",
+	})
+	if err != nil {
+		return
+	}
+	if len(pods.Items) != nodes {
+		return
+	}
+
+	for _, pod := range pods.Items {
+		if pod.Status.Phase != "Running" {
+			return
+		}
+	}
+
+	sidecarCheck = api.HealthcheckItem{Name: "sidecar", Status: api.HealthcheckStatusOK, Message: "sidecar service is running"}
+	return
+}
+
+func (c *ClusterK8sRunner) Healthcheck(fix bool, engine api.Engine, writer io.Writer) (*api.HealthcheckReport, error) {
+
+	// TODO: initialize the runner correctly.
+	// When the runner is unitialized, that is, when it has a nil pointer where one is needed.
+	// I considered changing DoHealthCheck to pass an input parameter, such as we do for DoRun and
+	// DoCollectOutputs, but there are already other TODO's in this file that makes me think
+	// initRunner needs to be overhauled. So until such time that there is better initialization,
+	// handle, I'll handle uninitialized runners by sending an empty report here.
+	// What this means is if the first thing you do is run the health check, you will get an empty
+	// report, but for subsequent times, the report is real.
+	report := api.HealthcheckReport{}
+
+	if c.pool != nil {
+		report.Checks = []api.HealthcheckItem{
+			c.healthcheckK8s(),
+			c.healthcheckEFS(),
+			c.healthcheckRedis(),
+			c.healthcheckSidecar(),
+		}
+	}
+
+	if !fix {
+		return &report, nil
+	}
+
+	// TODO: implement fix
+	return &report, nil
```

Comment:
Could you add a check such that if any of the checks are failed, we populate the fixes slice indicating that the pertinent fix was _omitted_ with message "unimplemented"?

Reply:
Added  the not implemented message

Now if everything is actually working, it returns like this, with or without the `-fix` flag:
```
checking runner cluster:k8s
finished checking runner cluster:k8s
Checks:
- k8s: ok; k8s cluster is running
- efs: ok; efs provisioner is running
- redis: ok; redis service is running
- sidecar: ok; sidecar service is running
No fixes applied.
```

If I kubectl delete the sidecar resources, I get an expected failure:

```
checking runner cluster:k8s
finished checking runner cluster:k8s
Checks:
- k8s: ok; k8s cluster is running
- efs: ok; efs provisioner is running
- redis: ok; redis service is running
- sidecar: failed; sidecar service is not reachable
No fixes applied.
```

And with the `-fix` flag applied, it shows the message.
```
checking runner cluster:k8s
finished checking runner cluster:k8s
Checks:
- k8s: ok; k8s cluster is running
- efs: ok; efs provisioner is running
- redis: ok; redis service is running
- sidecar: failed; sidecar service is not reachable
Fixes:
- sidecar: failed; Fix not implemented yet for this check.

```